demo/sieve/src/main.rs

```diff
@@ -0,0 +1,149 @@
+//! Sieve of Eratosthenes
+//!
+//! A sieve finds prime numbers by first assuming all candidates are prime,
+//! then progressively using small primes to mark their multiples composite.
+//!
+//! Note that a given prime p only needs to start marking its multiples >=p²,
+//! as anything less will already have been marked by an even smaller prime
+//! factor.  This also means that sieving up to N only requires primes up to
+//! sqrt(N) - a useful fact to prepare for parallel sieving.
+//!
+//! Here we have three forms of the sieve:
+//!
+//! - `sieve_serial`: direct iteration in serial
+//! - `sieve_chunks`: chunked iteration, still serial
+//! - `sieve_parallel`: chunked iteration in parallel
+//!
+//! The performance improvement of `sieve_chunks` is largely due to cache
+//! locality.  Since the sieve has to sweep repeatedly over the same memory,
+//! it's helpful that the chunk size can fit entirely in the CPU cache.  Then
+//! `sieve_parallel` also benefits from this as long as there's cache room for
+//! multiple chunks, for the separate jobs in each thread.
+
+extern crate itertools;
+extern crate rayon;
+extern crate time;
+
+use itertools::StrideMut;
+use rayon::prelude::*;
+
+const CHUNK_SIZE: usize = 100_000;
```

Comment:
Btw, it actually makes sense to me now. That's the main idea, you want to get chunks from RAM as rare as you could. So, basically, if you know your L2/L3 cache, you want to get the biggest chunk possible that can fit in there(taking in consideration that there are also other things happening on the system that also uses the cache).